File: i3/.i3/scripts/new_workspace.js
var exec = require('child_process').exec;

exec('i3-msg -t get_workspaces', function (error, stdout, stderr) {

  var workspaces = JSON.parse(stdout),
      workspace_numbers = [],
      lowest = 1;

  workspaces.forEach(function (x) { workspace_numbers.push(x.num); });

  while (workspace_numbers.indexOf(lowest) !== -1){ lowest++; }

  exec('i3-msg workspace ' + lowest);

});

File: i3/.i3/scripts/new_workspace.py
import json
import subprocess

I3MSG = '/usr/bin/i3-msg'

result = subprocess.Popen([I3MSG, '-t', 'get_workspaces'], stdout=subprocess.PIPE, stdin=subprocess.PIPE)
result = result.communicate()[0]
data = json.loads(result)

numbers = map(lambda x: x['num'], data)

lowest = 1

while lowest in numbers:
    lowest = lowest + 1

subprocess.call([I3MSG, 'workspace', str(lowest)])
